soccer/MainWindow.cpp

```diff
@@ -1082,47 +1089,56 @@ void MainWindow::on_actionSeed_triggered() {
 
 // Log controls
 void MainWindow::on_logHistoryLocation_sliderMoved(int value) {
-    // update current frame
-    int minFrame = _processor->logger().firstFrameNumber();
-    int maxFrame = _processor->logger().lastFrameNumber();
-    _doubleFrameNumber = value + minFrame;
-    _doubleFrameNumber = min<double>(maxFrame, _doubleFrameNumber);
-
-    emit historyLocationChanged(_doubleFrameNumber - minFrame);
+    // Sync frameNumber with logHistory slider
+    _doubleFrameNumber = value;
 
     // pause playback
-    live(false);
     _playbackRate = 0;
 }
 
+void MainWindow::on_logHistoryLocation_sliderPressed() {
+    on_logHistoryLocation_sliderMoved(_ui.logHistoryLocation->value());
+}
+
+void MainWindow::on_logHistoryLocation_sliderReleased() {
+    on_logHistoryLocation_sliderPressed();
+}
+
 void MainWindow::on_logPlaybackRewind_clicked() {
-    live(false);
-    _playbackRate += -0.5;
+    if (live()) {
+        _playbackRate = -1;
+    } else {
+        *_playbackRate += -0.5;
```

Comment:
oh, the equals method is overwritten for boost option. The += isn't. This is correct.